src/main/java/eu/darkbot/api/managers/HeroItemsAPI.java

```diff
@@ -1,136 +1,46 @@
      */
-    default Optional<Item> findItem(String itemId) {
-        return filterItem(id -> id.equals(itemId));
-    }
+    Optional<Item> getItem(@NotNull SelectableItem selectableItem);
 
     /**
-     * Search {@link Category} for given {@code itemId}.
-     *
-     * @param category to be searched
-     * @param itemId   to be looked for in given category
-     * @return first encounter of given item id
+     * Returns {@link Collection<Item>} of given {@link ItemCategory}
      */
-    default Optional<Item> findItem(@NotNull HeroItemsAPI.Category category, String itemId) {
-        return filterItem(category, id -> id.equals(itemId));
-    }
-
-    default Optional<Item> findItem(@NotNull SelectableItem selectableItem) {
-        Category category = selectableItem.getCategory();
-        return category == null ? findItem(selectableItem.getId()) :
-                findItem(category, selectableItem.getId());
-    }
+    Collection<? extends Item> getItems(@NotNull ItemCategory itemCategory);
 
     /**
-     * Filters items in every {@link Category}
+     * This method checks if given {@link Item} can be used in-game.
+     * If the item isn't in any of the action bars it may not be selectable.
      *
-     * @param filter the items with
-     * @return filtered {@link Optional<Item>} or {@link Optional#empty()}
+     * @param item to check
+     * @return non empty optional if item can be used
      */
-    default Optional<Item> filterItem(Predicate<String> filter) {
-        for (Category category : Category.values()) {
-            if (!hasCategory(category)) continue;
-
-            Optional<Item> item = filterItem(category, filter);
-            if (item.isPresent()) return item;
-        }
-        return Optional.empty();
-    }
+    Optional<Item> getAvailable(@NotNull SelectableItem item);
 
     /**
-     * Filters items in given {@link Category}
+     * Will try to use given {@link SelectableItem} with optional additional {@link ItemUseFlag}s.
+     * API should check {@link Item#isAvailable()} by default,
+     * and return {@link ItemUseResult#NOT_AVAILABLE} if isn't available.
      *
-     * @param category to be searched
-     * @param filter   the items with
-     * @return filtered {@link Optional<Item>} or {@link Optional#empty()}
-     */
-    default Optional<Item> filterItem(@NotNull HeroItemsAPI.Category category,
-                                      @NotNull Predicate<String> filter) {
-        if (!hasCategory(category))
-            return Optional.empty();
-
-        for (Item item : getItems(category))
-            if (filter.test(item.getId()))
-                return Optional.of(item);
-
-        return Optional.empty();
-    }
-
-    /**
-     * Represents all available categories of {@link HeroItemsAPI} items.
+     * @param selectableItem to be used
+     * @param itemFlags      which this method must respect
+     * @return use result of the selectableItem
```

Comment:
These explanations are again a bit poor

"API should check ..." should api check? are you talking to implementors or users? Make sure these javadocs are aimed at users for the API, not implementors

selectable item has bad wording, don't mix the name of the parameter with the text as if it was part of the same sentence, it just looks odd, this is done in a few other places

itemFlags are not explained at all here "which this method must respect" explains nothing
in the return a link to selectable item result would be helpful for the longer explanation